core/build.gradle

```diff
@@ -173,45 +173,63 @@ if (!rootProject.hasProperty('noWeb')) {
 // Run HttpServerStreamingTest separately with memory constraint.
 tasks.test.exclude '**/HttpServerStreamingTest**'
 tasks.shadedTest.exclude '**/HttpServerStreamingTest**'
-task testStreaming(type: Test,
-                   group: 'Verification',
-                   description: 'Runs the streaming tests.',
-                   dependsOn: tasks.shadedTestClasses) {
-    // Use small heap for streaming tests to quickly ensure we can stream the content larger than heap.
-    maxHeapSize = '64m'
-
-    include '**/HttpServerStreamingTest**'
-    testClassesDirs = tasks.shadedTest.testClassesDirs
-    classpath = testClassesDirs
-
-    // Set the class path as late as possible so that the 'shadedTest' task has the correct classpath.
-    doFirst {
-        classpath += project.files(configurations.shadedTestRuntime.resolve())
+testing {
+    suites {
+        testStreaming(JvmTestSuite) {
+
+            group = 'Verification'
+            description = 'Runs the streaming tests.'
+
+            targets {
+                all {
+                    testTask.configure {
+                        dependsOn(tasks.shadedTestClasses)
+                        // Use small heap for streaming tests to quickly ensure we can stream the content larger than heap.
+                        maxHeapSize = '64m'
+
+                        include '**/HttpServerStreamingTest**'
+                        testClassesDirs = tasks.shadedTest.testClassesDirs
+                        classpath = testClassesDirs
+
+                        // Set the class path as late as possible so that the 'shadedTest' task has the correct classpath.
+                        doFirst {
+                            classpath += project.files(configurations.shadedTestRuntime.resolve())
+                        }
+                    }
+                }
+            }
+        }
     }
 }
-tasks.shadedTest.finalizedBy tasks.testStreaming
-tasks.check.dependsOn tasks.testStreaming
+tasks.shadedTest.finalizedBy testing.suites.testStreaming
+tasks.check.dependsOn testing.suites.testStreaming
 
+// Run the test cases based on reactive-streams-tck only with non-flaky mode
 if (rootProject.findProperty('FLAKY_TESTS') != 'true') {
-    // Run the test cases based on reactive-streams-tck only with non-flaky mode
-    task testNg(type: Test,
-            group: 'Verification',
-            description: 'Runs the TestNG unit tests.',
-            dependsOn: tasks.shadedTestClasses) {
-        useTestNG()
```

Comment:
Gradle prohibits overriding the test engine with `Test` tasks.
On the other hand, the [JVM Test Suite Plugin](https://docs.gradle.org/current/userguide/jvm_test_suite_plugin.html#jvm_test_suite_plugin) supports multiple test engines.
So the test tasks have been migrated to the new test plugin.